R/methods.R

```diff
@@ -101,10 +101,14 @@ print.lints <- function(x, ...) {
     if (isTRUE(settings$error_on_lint)) {
       quit("no", 31L, FALSE) # nocov
     }
-  } else if (use_rstudio_source_markers) {
-    # Empty lints: clear RStudio source markers
-    rstudio_source_markers(x)
+  } else {
+    # Empty lints
+    cli_inform(c(i = "No lints found."))
```

Comment:
this is a classed signal:

```r
class(tryCatch(cli::cli_inform("xxx"), condition=identity))
# [1] "rlang_message" "message"       "condition"
```

Is that appropriate? I might have expected just `cat()` here, I guess {cli} adds nice aesthetics -- are there any analogues for "plain text to terminal"?

(or perhaps a message _is_ appropriate, WDYT?)

Reply:
Yes, we could use `cat()`, but I also think there are benefits to having this as a classed signal over `cat()` (e.g. the messages will show up in logs).

If we don't want a classed signal, either `cli_alert()` or its sibling should work:

``` r
cli::cli_alert_success("No lints found")
#> ✔ No lints found

class(tryCatch(cli::cli_alert_success("No lints found"), condition=identity))
#> [1] "cli_message" "condition"
```

<sup>Created on 2024-08-05 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

So we can have our pretty aesthetics cake and eat it too.